internal/server/auth/method/github/server.go

```diff
@@ -0,0 +1,166 @@
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"go.flipt.io/flipt/errors"
+	"go.flipt.io/flipt/internal/config"
+	storageauth "go.flipt.io/flipt/internal/storage/auth"
+	"go.flipt.io/flipt/rpc/flipt/auth"
+	"go.uber.org/zap"
+	"golang.org/x/oauth2"
+	oauth2GitHub "golang.org/x/oauth2/github"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const (
+	githubUserAPI = "https://api.github.com/user"
+)
+
+// OAuth2Client is our abstraction of communication with an OAuth2 Provider.
+type OAuth2Client interface {
+	AuthCodeURL(state string, opts ...oauth2.AuthCodeOption) string
+	Exchange(ctx context.Context, code string, opts ...oauth2.AuthCodeOption) (*oauth2.Token, error)
+	Client(ctx context.Context, t *oauth2.Token) *http.Client
+}
+
+const (
+	storageMetadataGithubEmail = "io.flipt.auth.github.email"
+	storageMetadataGithubName  = "io.flipt.auth.github.name"
+)
+
+// Server is an Github server side handler.
+type Server struct {
+	logger       *zap.Logger
+	store        storageauth.Store
+	config       config.AuthenticationConfig
+	oauth2Config OAuth2Client
+
+	auth.UnimplementedAuthenticationMethodGithubServiceServer
+}
+
+// NewServer constructs a Server.
+func NewServer(
+	logger *zap.Logger,
+	store storageauth.Store,
+	config config.AuthenticationConfig,
+) *Server {
+	return &Server{
+		logger: logger,
+		store:  store,
+		config: config,
+		oauth2Config: &oauth2.Config{
+			ClientID:     config.Methods.Github.Method.ClientId,
+			ClientSecret: config.Methods.Github.Method.ClientSecret,
+			Endpoint:     oauth2GitHub.Endpoint,
+			RedirectURL:  callbackURL(config.Methods.Github.Method.RedirectAddress),
+			Scopes:       []string{strings.Join(config.Methods.Github.Method.Scopes, ":")},
```

Comment:
This seems wrong to me. This feels like we're combining not the full scopes, but the noun and verb parts of a Github scope.
For example, how would the user configure both `user:email` and `read:user`? This sticks all the parts in the slice together with `:`.

I think we should just pass in the slice of scopes as is. Then the configurer has to do e.g. `scopes: ["user:email", "read:user"]`.
```suggestion
			Scopes:      config.Methods.Github.Method.Scopes,
```

Reply:
Got it. It would be up to the user to provide the scopes as is. I like that idea!